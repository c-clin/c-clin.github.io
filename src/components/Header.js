import React from 'react';
import { Element } from 'react-scroll';

const Header = () => {
  return (
    <Element name="Header">
      <section className="Header">
        <div className="Header__text-box">
          <div className="Header__text-box--primary">
            <h1>Catherine Lin</h1>
          </div>
          <div className="Header__text-box--secondary">
            <p>Frontend Web Developer</p>
          </div>
          <hr />
          <div className="Header__social-links">
            <a
              href="mailto:catherine.lin.93@gmail.com"
              target="_blank"
              rel="noopener noreferrer"
            >
              <i
                className="fa fa-envelope-o icon previous"
                aria-hidden="true"
              />
            </a>
            <a
              href="https://linkedin.com/in/catherine-lin-81545896"
              target="_blank"
              rel="noopener noreferrer"
            >
              <i className="fa fa-linkedin icon previous" aria-hidden="true" />
            </a>
            <a
              href="https://github.com/c-clin"
              target="_blank"
              rel="noopener noreferrer"
            >
              <i className="fa fa-github icon" aria-hidden="true" />
            </a>
          </div>
        </div>
      </section>
    </Element>
  );
};

export default Header;
