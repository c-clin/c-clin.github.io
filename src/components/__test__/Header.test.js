import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header', () => {
  const header = shallow(<Header />);

  it('renders properly', () => {
    expect(header).toMatchSnapshot();
  });

  it('renders the name as the h1 text', () => {
    expect(header.find('h1').text()).toEqual('Catherine Lin');
  });

  it('renders 3 icons', () => {
    expect(header.find('i')).toHaveLength(3);
  });
});
