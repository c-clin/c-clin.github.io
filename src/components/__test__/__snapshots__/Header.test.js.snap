// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <section
        className="Header"
>
        <div
                className="Header__text-box"
        >
                <div
                        className="Header__text-box--primary"
                >
                        <h1>
                                Catherine Lin
                        </h1>
                </div>
                <div
                        className="Header__text-box--secondary"
                >
                        <p>
                                Frontend Web Developer
                        </p>
                </div>
                <hr />
                <div
                        className="Header__social-links"
                >
                        <a
                                href="mailto:catherine.lin.93@gmail.com"
                                rel="noopener noreferrer"
                                target="_blank"
                        >
                                <i
                                        aria-hidden="true"
                                        className="fa fa-envelope-o icon previous"
                                />
                        </a>
                        <a
                                href="https://linkedin.com/in/catherine-lin-81545896"
                                rel="noopener noreferrer"
                                target="_blank"
                        >
                                <i
                                        aria-hidden="true"
                                        className="fa fa-linkedin icon previous"
                                />
                        </a>
                        <a
                                href="https://github.com/c-clin"
                                rel="noopener noreferrer"
                                target="_blank"
                        >
                                <i
                                        aria-hidden="true"
                                        className="fa fa-github icon"
                                />
                        </a>
                </div>
        </div>
</section>,
      "name": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Header__text-box"
>
          <div
                    className="Header__text-box--primary"
          >
                    <h1>
                              Catherine Lin
                    </h1>
          </div>
          <div
                    className="Header__text-box--secondary"
          >
                    <p>
                              Frontend Web Developer
                    </p>
          </div>
          <hr />
          <div
                    className="Header__social-links"
          >
                    <a
                              href="mailto:catherine.lin.93@gmail.com"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <i
                                        aria-hidden="true"
                                        className="fa fa-envelope-o icon previous"
                              />
                    </a>
                    <a
                              href="https://linkedin.com/in/catherine-lin-81545896"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <i
                                        aria-hidden="true"
                                        className="fa fa-linkedin icon previous"
                              />
                    </a>
                    <a
                              href="https://github.com/c-clin"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              <i
                                        aria-hidden="true"
                                        className="fa fa-github icon"
                              />
                    </a>
          </div>
</div>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Header__text-box--primary"
>
              <h1>
                            Catherine Lin
              </h1>
</div>,
            <div
              className="Header__text-box--secondary"
>
              <p>
                            Frontend Web Developer
              </p>
</div>,
            <hr />,
            <div
              className="Header__social-links"
>
              <a
                            href="mailto:catherine.lin.93@gmail.com"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <i
                                          aria-hidden="true"
                                          className="fa fa-envelope-o icon previous"
                            />
              </a>
              <a
                            href="https://linkedin.com/in/catherine-lin-81545896"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <i
                                          aria-hidden="true"
                                          className="fa fa-linkedin icon previous"
                            />
              </a>
              <a
                            href="https://github.com/c-clin"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            <i
                                          aria-hidden="true"
                                          className="fa fa-github icon"
                            />
              </a>
</div>,
          ],
          "className": "Header__text-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Catherine Lin
</h1>,
              "className": "Header__text-box--primary",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Catherine Lin",
              },
              "ref": null,
              "rendered": "Catherine Lin",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Frontend Web Developer
</p>,
              "className": "Header__text-box--secondary",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Frontend Web Developer",
              },
              "ref": null,
              "rendered": "Frontend Web Developer",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="mailto:catherine.lin.93@gmail.com"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <i
                                    aria-hidden="true"
                                    className="fa fa-envelope-o icon previous"
                  />
</a>,
                <a
                  href="https://linkedin.com/in/catherine-lin-81545896"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <i
                                    aria-hidden="true"
                                    className="fa fa-linkedin icon previous"
                  />
</a>,
                <a
                  href="https://github.com/c-clin"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  <i
                                    aria-hidden="true"
                                    className="fa fa-github icon"
                  />
</a>,
              ],
              "className": "Header__social-links",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-envelope-o icon previous"
/>,
                  "href": "mailto:catherine.lin.93@gmail.com",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-envelope-o icon previous",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-linkedin icon previous"
/>,
                  "href": "https://linkedin.com/in/catherine-lin-81545896",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-linkedin icon previous",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-github icon"
/>,
                  "href": "https://github.com/c-clin",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-github icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <section
          className="Header"
>
          <div
                    className="Header__text-box"
          >
                    <div
                              className="Header__text-box--primary"
                    >
                              <h1>
                                        Catherine Lin
                              </h1>
                    </div>
                    <div
                              className="Header__text-box--secondary"
                    >
                              <p>
                                        Frontend Web Developer
                              </p>
                    </div>
                    <hr />
                    <div
                              className="Header__social-links"
                    >
                              <a
                                        href="mailto:catherine.lin.93@gmail.com"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        <i
                                                  aria-hidden="true"
                                                  className="fa fa-envelope-o icon previous"
                                        />
                              </a>
                              <a
                                        href="https://linkedin.com/in/catherine-lin-81545896"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        <i
                                                  aria-hidden="true"
                                                  className="fa fa-linkedin icon previous"
                                        />
                              </a>
                              <a
                                        href="https://github.com/c-clin"
                                        rel="noopener noreferrer"
                                        target="_blank"
                              >
                                        <i
                                                  aria-hidden="true"
                                                  className="fa fa-github icon"
                                        />
                              </a>
                    </div>
          </div>
</section>,
        "name": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Header__text-box"
>
            <div
                        className="Header__text-box--primary"
            >
                        <h1>
                                    Catherine Lin
                        </h1>
            </div>
            <div
                        className="Header__text-box--secondary"
            >
                        <p>
                                    Frontend Web Developer
                        </p>
            </div>
            <hr />
            <div
                        className="Header__social-links"
            >
                        <a
                                    href="mailto:catherine.lin.93@gmail.com"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <i
                                                aria-hidden="true"
                                                className="fa fa-envelope-o icon previous"
                                    />
                        </a>
                        <a
                                    href="https://linkedin.com/in/catherine-lin-81545896"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <i
                                                aria-hidden="true"
                                                className="fa fa-linkedin icon previous"
                                    />
                        </a>
                        <a
                                    href="https://github.com/c-clin"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    <i
                                                aria-hidden="true"
                                                className="fa fa-github icon"
                                    />
                        </a>
            </div>
</div>,
          "className": "Header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Header__text-box--primary"
>
                <h1>
                                Catherine Lin
                </h1>
</div>,
              <div
                className="Header__text-box--secondary"
>
                <p>
                                Frontend Web Developer
                </p>
</div>,
              <hr />,
              <div
                className="Header__social-links"
>
                <a
                                href="mailto:catherine.lin.93@gmail.com"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <i
                                                aria-hidden="true"
                                                className="fa fa-envelope-o icon previous"
                                />
                </a>
                <a
                                href="https://linkedin.com/in/catherine-lin-81545896"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <i
                                                aria-hidden="true"
                                                className="fa fa-linkedin icon previous"
                                />
                </a>
                <a
                                href="https://github.com/c-clin"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                <i
                                                aria-hidden="true"
                                                className="fa fa-github icon"
                                />
                </a>
</div>,
            ],
            "className": "Header__text-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Catherine Lin
</h1>,
                "className": "Header__text-box--primary",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Catherine Lin",
                },
                "ref": null,
                "rendered": "Catherine Lin",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Frontend Web Developer
</p>,
                "className": "Header__text-box--secondary",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Frontend Web Developer",
                },
                "ref": null,
                "rendered": "Frontend Web Developer",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="mailto:catherine.lin.93@gmail.com"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <i
                                        aria-hidden="true"
                                        className="fa fa-envelope-o icon previous"
                    />
</a>,
                  <a
                    href="https://linkedin.com/in/catherine-lin-81545896"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <i
                                        aria-hidden="true"
                                        className="fa fa-linkedin icon previous"
                    />
</a>,
                  <a
                    href="https://github.com/c-clin"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    <i
                                        aria-hidden="true"
                                        className="fa fa-github icon"
                    />
</a>,
                ],
                "className": "Header__social-links",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-envelope-o icon previous"
/>,
                    "href": "mailto:catherine.lin.93@gmail.com",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-envelope-o icon previous",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-linkedin icon previous"
/>,
                    "href": "https://linkedin.com/in/catherine-lin-81545896",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-linkedin icon previous",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-github icon"
/>,
                    "href": "https://github.com/c-clin",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-github icon",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
